aten/src/THC/THCNumerics.cuh

```diff
@@ -364,8 +364,25 @@ struct THCNumerics<at::BFloat16> {
   static inline __host__ __device__  at::BFloat16 sub  (at::BFloat16 a, at::BFloat16 b) { return a - b; }
   static inline __host__ __device__  at::BFloat16 pow  (at::BFloat16 a, at::BFloat16 b) { return powf(a, b); }
   static inline __host__ __device__  at::BFloat16 atan2(at::BFloat16 a, at::BFloat16 b) { return atan2f(a, b); }
-  static inline __host__ __device__  bool isnan(at::BFloat16 a) { return ::isnan(a); }
-  static inline __host__ __device__  bool isinf(at::BFloat16 a) { return ::isinf(a); }
+  static inline __host__ __device__ bool isnan(at::BFloat16 a) {
+    #ifdef _MSC_VER
+      // Windows requires this explicit conversion. The reason is unclear
+      // related issue with clang: https://reviews.llvm.org/D37906
+      return ::isnan((float) a);
+    #else
+      return ::isnan(a);
+    #endif
+  }
+
+  static inline __host__ __device__ bool isinf(at::BFloat16 a) {
+    #ifdef _MSC_VER
```

Comment:
Would it be possible to always explicitly convert and remove these `#ifdef`s?

Reply:
Ideally we want to avoid conversion due to performance reasons but Windows requires us to do this.

Reply:
Is there an `isnan` specifically for bfloat? I looked up the codebase and couldn't find one. If there is an `isnan` specifically made for bfloat, then I agree with you. Otherwise even if the compilation passes without an explicit conversion, an implicit one is always inserted by the compiler. That's why I'm asking.